docs/decisions/0003-instructor-dashboard-rest-api-architecture.rst

```diff
@@ -0,0 +1,276 @@
+| Method | Endpoint                                                                           | Services                                | Status    | Existing API Path                                        |
++========+====================================================================================+=========================================+===========+==========================================================+
+| GET    | /api/instructor/v1/courses/{course_key}/discussions/topics                         | discussion                              | Missing   | N/A                                                      |
++--------+------------------------------------------------------------------------------------+-----------------------------------------+-----------+----------------------------------------------------------+
+| POST   | /api/instructor/v1/courses/{course_key}/discussions/topics                         | discussion, xmodule.modulestore         | Missing   | N/A                                                      |
++--------+------------------------------------------------------------------------------------+-----------------------------------------+-----------+----------------------------------------------------------+
+| PUT    | /api/instructor/v1/courses/{course_key}/discussions/topics/{topic_id}              | discussion                              | Missing   | N/A                                                      |
++--------+------------------------------------------------------------------------------------+-----------------------------------------+-----------+----------------------------------------------------------+
+| DELETE | /api/instructor/v1/courses/{course_key}/discussions/topics/{topic_id}              | discussion                              | Missing   | N/A                                                      |
++--------+------------------------------------------------------------------------------------+-----------------------------------------+-----------+----------------------------------------------------------+
+| GET    | /api/instructor/v1/courses/{course_key}/discussions/settings                       | discussion                              | Missing   | N/A                                                      |
++--------+------------------------------------------------------------------------------------+-----------------------------------------+-----------+----------------------------------------------------------+
+| PUT    | /api/instructor/v1/courses/{course_key}/discussions/settings                       | discussion                              | Missing   | N/A                                                      |
++--------+------------------------------------------------------------------------------------+-----------------------------------------+-----------+----------------------------------------------------------+
+
+Extensions & Deadlines API
```

Comment:
Course have two related endpoints, we can probably add more there as needed to keep this code localized: https://github.com/openedx/edx-platform/blob/master/docs/lms-openapi.yaml#L801